src/services/item/plugins/embeddedLink/service.ts

```diff
@@ -40,27 +58,100 @@ const CSP_FRAME_NONE = ["frame-ancestors 'none'", "frame-ancestors 'self'"];
+  }
+
   private assertUrlIsValid(url: string) {
     if (!isValidUrl(url)) {
       throw new InvalidUrl(url);
     }
   }
 
-  async getLinkMetadata(iframelyHrefOrigin: string, url: string): Promise<LinkMetadata> {
+  async getLinkMetadata(url: string): Promise<LinkMetadata> {
     this.assertUrlIsValid(url);
+    try {
+      const response = await fetch(
+        `${this.iframelyHrefOrigin}/iframely?uri=${encodeURIComponent(url)}`,
+      );
+
+      // better clues on how to extract the metadata here: https://iframely.com/docs/links
+      const { meta = {}, html, links = [] } = (await response.json()) as IframelyResponse;
+      const { title, description } = meta;
+
+      return {
+        title: title?.trim(),
+        description: description?.trim(),
+        html,
+        thumbnails: links.filter(({ rel }) => hasThumbnailRel(rel)).map(({ href }) => href),
+        icons: links
+          .filter(({ rel }: { rel: string[] }) => hasIconRel(rel))
+          .map(({ href }) => href),
+      };
+    } catch (e) {
+      // do not fail on iframely error
+      // send error to sentry
+      captureException(e);
+      console.error(e);
+      return { icons: [], thumbnails: [], description: '', title: '', html: '' };
+    }
+  }
 
-    const response = await fetch(`${iframelyHrefOrigin}/iframely?uri=${encodeURIComponent(url)}`);
-    // better clues on how to extract the metadata here: https://iframely.com/docs/links
-    const { meta = {}, html, links = [] } = (await response.json()) as IframelyResponse;
-    const { title, description } = meta;
+  /**
+   * Create link extra object given url and initial data
+   * @param url new url to get metadata from
+   * @param itemExtra  initial link extra
+   * @returns valid link extra object
+   */
+  private async createExtra(
+    url: string,
+    itemExtra?: LinkItemExtraProperties,
+  ): Promise<LinkItemExtraProperties> {
+    // get metadata for empty extra or new url
+    let metadata = {};
+    if (!itemExtra || url !== itemExtra.url) {
+      metadata = await this.getLinkMetadata(url);
+    }
+    return {
+      ...itemExtra,
+      ...metadata,
+      url,
+    };
+  }
 
+  /**
+   * Create a valid link embedded item object
+   * @param item initial item properties
+   * @param data link properties
+   * @returns complete link object
+   */
+  private async createLink(
+    item: Partial<EmbeddedLinkItem> & Pick<EmbeddedLinkItem, 'name'>,
+    settings: {
+      showLinkIframe?: boolean;
+      showLinkButton?: boolean;
+    },
+    extra: LinkItemExtra,
+  ) {
     return {
-      title: title?.trim(),
-      description: description?.trim(),
-      html,
-      thumbnails: links.filter(({ rel }) => hasThumbnailRel(rel)).map(({ href }) => href),
-      icons: links.filter(({ rel }: { rel: string[] }) => hasIconRel(rel)).map(({ href }) => href),
+      ...item,
+      type: ItemType.LINK,
+      extra,
+      settings: {
+        ...(item.settings ?? {}),
```

Comment:
It is allowed to spread `undefined`, I don't think you need to use a nullish coalescing operator here.
```suggestion
        ...item.settings,
```

Reply:
It is not allowed to spread `undefined` tho??? It is not an iterable.